boskos/ranch/ranch_test.go

```diff
@@ -202,88 +239,98 @@ func TestAcquireRoundRobin(t *testing.T) {
+			resName:     "res",
+			owner:       "user",
+			dest:        "d",
+			expectErr:   &OwnerNotMatch{"merlin", "user"},
+			expectedRes: common.NewResource("res", "t", "s", "merlin", startTime),
 		},
 		{
-			name: "no match name",
-			resources: []common.Resource{
-				common.NewResource("foo", "t", "s", "merlin", FakeNow),
-			},
-			resName:   "res",
-			owner:     "user",
-			dest:      "d",
-			expectErr: &ResourceNotFound{"res"},
+			name:        "no match name",
+			resource:    common.NewResource("foo", "t", "s", "merlin", startTime),
+			resName:     "res",
+			owner:       "user",
+			dest:        "d",
+			expectErr:   &ResourceNotFound{"res"},
+			expectedRes: common.Resource{},
 		},
 		{
-			name: "ok",
-			resources: []common.Resource{
-				common.NewResource("res", "t", "s", "merlin", FakeNow),
+			name:        "ok",
+			resource:    common.NewResource("res", "t", "s", "merlin", startTime),
+			resName:     "res",
+			owner:       "merlin",
+			dest:        "d",
+			expectErr:   nil,
+			expectedRes: common.NewResource("res", "t", "d", "", fakeNow),
+		},
+		{
+			name:     "ok - has dynamic resource lf no lifespan",
+			resource: common.NewResource("res", "t", "s", "merlin", startTime),
+			dResource: common.DynamicResourceLifeCycle{
+				Type: "t",
 			},
-			resName:   "res",
-			owner:     "merlin",
-			dest:      "d",
-			expectErr: nil,
+			resName:     "res",
+			owner:       "merlin",
+			dest:        "d",
+			expectErr:   nil,
+			expectedRes: common.NewResource("res", "t", "d", "", fakeNow),
+		},
+		{
+			name:     "ok - has dynamic resource lf with lifespan",
+			resource: common.NewResource("res", "t", "s", "merlin", startTime),
+			dResource: common.DynamicResourceLifeCycle{
+				Type:     "t",
+				LifeSpan: &lifespan,
+			},
+			resName:     "res",
+			owner:       "merlin",
+			dest:        "d",
+			expectErr:   nil,
+			expectedRes: updatedRes,
 		},
 	}
 
 	for _, tc := range testcases {
-		c := MakeTestRanch(tc.resources)
-		err := c.Release(tc.resName, tc.dest, tc.owner)
-		if !AreErrorsEqual(err, tc.expectErr) {
-			t.Errorf("%s - Got error %v, expect error %v", tc.name, err, tc.expectErr)
+		c := MakeTestRanch([]common.Resource{tc.resource}, []common.DynamicResourceLifeCycle{tc.dResource})
+		releaseErr := c.Release(tc.resName, tc.dest, tc.owner)
+		if !AreErrorsEqual(releaseErr, tc.expectErr) {
+			t.Errorf("%s - Got error %v, expectedRes error %v", tc.name, releaseErr, tc.expectErr)
 			continue
 		}
-		resources, err2 := c.Storage.GetResources()
-		if err2 != nil {
-			t.Errorf("failed to get resources")
-			continue
-		}
-		if err == nil {
-			if resources[0].Owner != "" {
-				t.Errorf("%s - Wrong owner after release. Got %v, expect empty", tc.name, resources[0].Owner)
-			} else if resources[0].State != tc.dest {
-				t.Errorf("%s - Wrong state after release. Got %v, expect %v", tc.name, resources[0].State, tc.dest)
-			} else if !resources[0].LastUpdate.After(FakeNow) {
```

Comment:
eh - seems we dropped this assert?

Reply:
I changed the test to compare resources, and since we use our own time.now impl, we know what to expect.